### ISSUE TITLE
Transpilation of for ... of iteration doesn't work with built-in iterables

### ISSUE THREAD

**Issue by rictic** [google/closure-compiler#929]
[View this example in the Debugger](https://closure-compiler-debugger.appspot.com/#input0%3Dlet%2520map%2520%253D%2520new%2520Map()%253B%250Amap.set('a'%252C%25201)%253B%250Amap.set('b'%252C%25202)%253B%250Amap.set('c'%252C%25203)%253B%250A%250Afor%2520(var%2520x%2520of%2520map)%2520%257B%250A%2520%2520console.log(x)%253B%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1)

This code works fine uncompiled in Chrome 42 and compiles with no warnings:
```javascript
let map = new Map();
map.set('a', 1);
map.set('b', 2);
map.set('c', 3);

for (var x of map) {
  console.log(x);
}
```

However when compiled the map is run through $jscomp.makeIterator which assumes either an array, a string, or an object with an "$$iterator" property, and throws an error.

**Comment by rictic** [comment]
Oh, it looks like the [latest version of es6_runtime does the right thing here](https://github.com/google/closure-compiler/blob/62b5a5d913f947d0d5fb6a5381884209db202785/src/com/google/javascript/jscomp/js/es6_runtime.js#L52), so maybe this will be fixed in the next release?

**Comment by MatrixFrog** [comment]
Yup. Fixed in e2317ea07e42dfff97d8f3784f49ddfe0ce51602. We're hoping to get a release out in April, but since our time machine is currently broken, it will most likely be early next week instead.
